### PR TITLE
[Monitor] `az monitor log-analytics workspace`: Add `list-link-target` and `list-available-service-tier`

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

az monitor log-analytics workspace

**Description**

**Testing Guide**

**History Notes**

[Monitor] `az monitor log-analytics workspace`: Add `list-link-target` and `list-available-service-tier`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

related aaz https://github.com/Azure/aaz/pull/364

### CODE REVIEW

**Comment by yonzhan** [comment]
Monitor

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/__cmd_group.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "monitor log-analytics workspace",
 )
 class __CMDGroup(AAZCommandGroup):
-    """Manage Azure log analytics workspace
+    """monitor log-analytics workspace
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/__cmd_group.py:18]: Why the command group description changed?

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/available_service_tier/_list.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "monitor log-analytics workspace available-service-tier list",
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/available_service_tier/_list.py:15]: According to the [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior), Recommend using the following name
```suggestion
    "monitor log-analytics workspace list-available-service-tier",
```

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/link_target/_list.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "monitor log-analytics workspace link-target list",
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/link_target/_list.py:15]: Same concern here

File: src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_list_link_target.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "monitor log-analytics workspace list-link-target",
+)
+class ListLinkTarget(AAZCommand):
+    """List a list of workspaces which the current user has administrator privileges and are not associated with an Azure Subscription. The subscriptionId parameter in the Url is ignored.
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/aaz/latest/monitor/log_analytics/workspace/_list_link_target.py:18]: The last line about subscription ID should be removed in help

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**